lib/std/bounded_array.zig

```diff
@@ -39,10 +40,10 @@ pub fn BoundedArrayAligned(
 ) type {
     return struct {
         const Self = @This();
-        const Len = std.math.IntFittingRange(0, buffer_capacity);
 
         buffer: [buffer_capacity]T align(alignment) = undefined,
-        len: Len = 0,
+        len: std.math.IntFittingRange(0, buffer_capacity) = 0,
+        comptime capacity: usize = buffer_capacity,
```

Comment:
Generally, things should not be default initialized to `undefined`, and struct fields should only be given defaults when those fields can be independently configured in a struct initialization without needing to remember other fields. This is not the case here. The set {undefined, 0} makes sense together but not separately, so instead there should just be a template init constant to copy:

```zig
pub const init: @This() = .{ .buffer = undefined, .len = 0 };
```

and then the initalization code looks like 

```zig
var a = BoundedArray(T, cap).init;
```

Reply:
I like that!